R/run_trial.R

```diff
@@ -151,46 +151,67 @@ run_trial <- function(
     }
 
     # post-processing to get evaluation metrics
-    if(isTRUE(design$evaluation$type %in% target_types_auc)) {
-      auc_eval <- calc_auc_from_regimen(
-        regimen = res$final_regimen,
-        parameters = pars_true_i, # true patient parameters
-        model = design$sim$model,
-        target_design = design$evaluation,
-        covariates = covs
-      )
-      eval_time <- get_sampling_times_from_scheme(
-        design$evaluation$scheme,
-        res$final_regimen
-      )
-      eval_exposure <- rbind(
-        eval_exposure,
-        data.frame(
-          id = i,
-          time = eval_time,
-          auc = auc_eval
-        )
-      )
-    } else if (design$evaluation$type %in% target_types_conc){
-      conc_eval <- calc_concentration_from_regimen(
-        regimen = res$final_regimen,
-        parameters = pars_true_i, # true patient parameters
-        model = design$sim$model,
-        target_design = design$evaluation,
-        covariates = covs
-      )
-      eval_time <- get_sampling_times_from_scheme(
-        design$evaluation$scheme,
-        res$final_regimen
-      )
-      eval_exposure <- rbind(
-        eval_exposure,
-        data.frame(
-          id = i,
-          time = eval_time,
-          conc = conc_eval
-        )
-      )
+    if (!is.null(design$evaluation)){
```

Comment:
this is a fairly contained block of code, and the function is already getting quite long. perhaps we can wrap this into a function that returns `eval_exposure`?

it also seems like we might be able to prevent having quite so much near-duplicated code if we do something like this somewhere in here:

```r
if (design_type %in% target_types_auc) {
  f_calc <- calc_auc_from_regimen
else if (design_type %in% target_types_conc) {
  f_calc <- calc_concentration_from_regimen 
}
new_evaluate_exposure <- f_calc(
  regimen = res$final_regimen,
  parameters = pars_true_i, # true patient parameters
  model = design$sim$model,
  target_design = tmp_eval_design,
  covariates = covs
)
```